File: test/fixtures/rust/small/entity_service_100.rs
// Rust Fixture (target ~100 lines)
// Generated for benchmark testing

use anyhow::Result;
use chrono::{DateTime, Utc};
use std::sync::Arc;

#[derive(Debug, Clone)]
pub struct Entity {
    pub id: u64,
    pub name: String,
    pub email: String,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone)]
pub struct CreateDto {
    pub name: String,
    pub email: String,
}

#[derive(Debug, Clone)]
pub struct UpdateDto {
    pub name: Option<String>,
    pub email: Option<String>,
    pub is_active: Option<bool>,
}

pub struct EntityService {
    db: Arc<Database>,
    logger: Arc<Logger>,
    cache: Arc<Cache<u64, Entity>>,
}

impl EntityService {
    pub fn new(db: Arc<Database>, logger: Arc<Logger>, cache: Arc<Cache<u64, Entity>>) -> Self {
        Self { db, logger, cache }
    }


    pub async fn operation_0(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_1(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_2(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }
}

File: test/fixtures/zig/small/entity_service_100.zig
// Zig Fixture (target ~100 lines)
// Generated for benchmark testing

const std = @import("std");

pub const Entity = struct {
    id: u64,
    name: []const u8,
    email: []const u8,
    is_active: bool,
    created_at: i64,
    updated_at: i64,
};

pub const CreateDto = struct {
    name: []const u8,
    email: []const u8,
};

pub const UpdateDto = struct {
    name: ?[]const u8 = null,
    email: ?[]const u8 = null,
    is_active: ?bool = null,
};

pub const EntityService = struct {
    const Self = @This();

    db: *Database,
    logger: *Logger,
    cache: *Cache(u64, Entity),
    allocator: std.mem.Allocator,

    pub fn init(allocator: std.mem.Allocator, db: *Database, logger: *Logger, cache: *Cache(u64, Entity)) !*Self {
        var self = try allocator.create(Self);
        self.* = .{
            .db = db,
            .logger = logger,
            .cache = cache,
            .allocator = allocator,
        };
        return self;
    }

    pub fn deinit(self: *Self) void {
        self.allocator.destroy(self);
    }

    pub fn operation0(self: *Self, id: u64, data: []const u8) !?Entity {
        const result = self.db.query("SELECT * FROM entities WHERE id = ?", .{id}) catch |err| {
            self.logger.err("Operation failed: {}", .{err});
            return err;
        };
        self.logger.debug("Fetched {}", .{id});
        return if (result) |r| Entity.fromRow(r) else null;
    }

    pub fn operation1(self: *Self, id: u64, data: []const u8) !?Entity {
        const result = self.db.query("SELECT * FROM entities WHERE id = ?", .{id}) catch |err| {
            self.logger.err("Operation failed: {}", .{err});
            return err;
        };
        self.logger.debug("Fetched {}", .{id});
        return if (result) |r| Entity.fromRow(r) else null;
    }

    pub fn operation2(self: *Self, id: u64, data: []const u8) !?Entity {
        const result = self.db.query("SELECT * FROM entities WHERE id = ?", .{id}) catch |err| {
            self.logger.err("Operation failed: {}", .{err});
            return err;
        };
        self.logger.debug("Fetched {}", .{id});
        return if (result) |r| Entity.fromRow(r) else null;
    }

    pub fn operation3(self: *Self, id: u64, data: []const u8) !?Entity {
        const result = self.db.query("SELECT * FROM entities WHERE id = ?", .{id}) catch |err| {
            self.logger.err("Operation failed: {}", .{err});
            return err;
        };
        self.logger.debug("Fetched {}", .{id});
        return if (result) |r| Entity.fromRow(r) else null;
    }
};
